server/src/domain/usecases/account.rs

```diff
@@ -0,0 +1,175 @@
+use crate::adapters::storage::cache::Cache;
+use crate::adapters::storage::Storages;
+use crate::domain::entities::account::{Account, BalanceResponse};
+use crate::domain::entities::transaction::Operation::{
+    Replenish, TransferDecrease, TransferIncrease, Withdraw,
+};
+use crate::domain::entities::transaction::{
+    Operation, Transaction, TransactionResponse, TransferRequest, TransferResponse,
+};
+use crate::domain::errors::AppError;
+use crate::domain::errors::AppError::{
+    AccountNotExists, Overdraft, SelfTransfer, ZeroValueTransaction,
+};
+use std::sync::{Arc, RwLock};
+
+/// Создание нового счета.
+pub fn new_account<S: Storages>(storage: Arc<RwLock<S>>) -> TransactionResponse {
+    // получение названия последнего счета
+    let last_name: String = storage.write().unwrap().db().get_last_account_name();
+
+    // создание нового счета
+    let mut account: Account = Account::new(last_name);
+    let acc_name: String = account.name.clone();
+
+    // создание транзакиции о создании счета
+    let tx_new: Transaction = Transaction::new(
+        0_u32,
+        Operation::default(),
+        f64::default(),
+        f64::default(),
+        f64::default(),
+    );
+    // добавление транзакции в список транзакций счета
+    account.transactions.push(tx_new);
+    // добавление счета в db
+    storage.write().unwrap().db().create_account(account);
+    // body
+    let tx: TransactionResponse = TransactionResponse::new(acc_name, 0_u32, 0_f64);
+    // backup
+    storage.write().unwrap().db().backup_store();
+
+    tx
+}
+
+/// Изменение баланса счета.
+pub fn change_acc_balance<S: Storages>(
+    storage: &Arc<RwLock<S>>,
+    trans_value: f64,
+    account_name: &String,
+    operation: Operation,
+) -> Result<TransactionResponse, AppError> {
+    // проверка наличия счета
+    if !storage.write().unwrap().db().check_key(account_name) {
+        return Err(AccountNotExists(account_name.to_string()));
+    }
+    // проверка на наличие изменение баланса на 0 или меньше
+    if trans_value <= 0_f64 {
+        return Err(ZeroValueTransaction);
+    }
+
+    let mut binding = storage.write().unwrap();
+    // получение счета
+    let cur_acc: &mut Account = binding.db().get_mut_account(account_name);
+    // проверка на снятие или перевод больше, чем есть на счете
+    if (operation == Withdraw || operation == TransferDecrease) && cur_acc.balance < trans_value {
```

Comment:
```suggestion
    if matches!(operation, Withdraw | TransferDecrease) && cur_acc.balance < trans_value {
```

Можно еще так писать, чтобы не дублировать `operation`

Reply:
done